engine/execution/ingestion/fetcher.go

```diff
@@ -0,0 +1,8 @@
+package ingestion
+
+import "github.com/onflow/flow-go/model/flow"
+
+type CollectionFetcher interface {
```

Comment:
Why is this not just a part of the other fetcher.go file?

Since this is a public interface, some comments would be nice, especially about the errors that are expected.